### ISSUE TITLE
Mac support / Builds?

### ISSUE THREAD

**Issue by trypsynth** [ars3niy/tdlib-purple#143]
I'm a fully blind Telegram user, and wanted to see if I could use this plugin accessibly (we currently have no accessible Telegram clients for the Mac). I'm using Adium as a client. Is there a way to build this for macOS, or have macOS builds released along with the Windows ones?
Thank you.

**Comment by BenWiederhake** [comment]
I am aware of a blind user that used Finch with telegram-purple with great success. Finch is a text-only interface to libpurple, just like Pidgin and Adium are GUI interfaces to libpurple. [Link to finch](https://pidgin.im/install/#finch) I guess Homebrew Pidgin also installs Finch? [Link to what seems to be the Homebrew Code that installs Pidgin and also seems to install Finch](https://github.com/Homebrew/homebrew-core/blob/master/Formula/pidgin.rb) I hope that tdlib-purple will work nicely for you.

I'm sorry, I'm not aware of any MacOS builds. :(

**Comment by trypsynth** [comment]
How would I install TDLib-purple when using Finch? If I can get it built, it'll probably work with Adium as well.

**Comment by BenWiederhake** [comment]
In essentially the same way; tdlib-purple installs itself by placing a `.so`-file in a specific folder. Finch either looks in the same folder, in which case it just works, or it looks in a different folder, in which case you can either just copy the file over, or even better we fix the install script.

**Comment by egslava** [comment]
Hello, any results, @TheQuinbox @BenWiederhake . Is it possible to get an already built *.AdiumLibpurplePlugin for tdlib-purple?

**Comment by BenWiederhake** [comment]
Absolutely! Just build them, send them to me, and I'll publish them to the release page. I don't provide any support for proprietary software.

I suggest using [my fork](https://github.com/BenWiederhake/tdlib-purple) as a starting point, as I've merged many of the PRs from this repository.
